allow export and display of attribute labels containing html special characters

rrennick opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Export attribute names containing html special characters without decoding,
Allow editing attribute names containing html special characters.

## Current Code
| Editor Entry | DB Value | Editor Display | Product Display | Exported CSV Field | Imported DB Field |
| ------------ | --------- | ------------- | ---------------- | -------------------- | ------------------ |
| `&quot;A3+&quot;` | `&quot;A3+&quot;` | `"A3+"` | `"A3+"` | `"""A3 """` | `"A3 "` |

## PR Applied
| Editor Entry | DB Value | Editor Display | Product Display | Exported CSV Field | Imported DB Field |
| ------------ | --------- | ------------- | ---------------- | -------------------- | ------------------ |
| `&quot;A3+&quot;` | `&quot;A3+&quot;` | `&quot;A3+&quot;` | `"A3+"` | `"&quot;A3+&quot;"` | `&quot;A3+&quot;` |

Closes #21457 .

### How to test the changes in this Pull Request:

1. Enter an attribute name with one or more html special characters
2. Save & Export
3. Review export file
4. Delete the attribute
5. Import the exported CSV

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Fix export and edit of attribute labels with html encoded special characters.

rrennick committed:
allow export and display of attribute labels containing html special characters

rrennick commented:
I'm not sure why this one is failing phpcs. I changed 1 line in `includes/admin/class-wc-admin-attributes.php` & getting 24 errors that are no where near the line that I edited.

claudiulodro commented:
phpcs scans the entire file for errors, and there are still plenty of areas that need to be brought up to phpcs compliance. Don't worry about phpcs errors that you didn't create. :)

claudiulodro reviewed (changes requested):
Thanks for the contribution. This seems to work slightly better than the old version. :)

The main issue I notice is that for the attributes I tested the `Attribute N value(s)` column gets double-quoted like so: `"'+A, A+, B'"`. This leads to the following attributes when imported:

<img width="248" alt="screen shot 2018-11-14 at 11 42 38 am" src="https://user-images.githubusercontent.com/7317227/48508194-cd7e8680-e802-11e8-841d-f63addafaf81.png">

If we can get that issue sorted, this will be good to merge.

rrennick commented:
@claudiulodro Can you provide the specific terms you added? I tested with

```shell
|      39 | AB+CD'"               | abcd                  |          0 |
|      40 | EF,GH                 | efgh                  |          0 |
```

I added both attributes to a product, exported, deleted terms, then imported. The DB values were the same & the CSV field was

```
"AB+CD'"", EF\,GH"
```
<img width="180" alt="screen shot 2018-11-15 at 11 50 28 am" src="https://user-images.githubusercontent.com/343847/48564410-b7d59380-e8cc-11e8-85cd-65fef6c1e662.png">

claudiulodro commented:
Sorry, it may not have been clear in my example. I tested with three global attributes: `+A`, `A+`, `B`. I'll investigate further; it may not be related to this change. I believe [`escape_date`](https://github.com/woocommerce/woocommerce/blob/c2da2afa6d810073eb55762dc0a1052bd0345ad1/includes/export/abstract-wc-csv-exporter.php#L353) is adding extra quotes since the column starts with a `+`.

rrennick commented:
>  I believe escape_date is adding extra quotes since the column starts with a +.

The first @see indicates it only needs to be prepended with the `'` rather than wrapped. In Excel, a cell starting with a `'` is treated as a literal string. We could fix this in a separate PR by removing the concatenation of the trailing `'` and adding a the reverse pattern match on import to remove the leading `'`.

claudiulodro reviewed (approved):
Looking further into this, it appears that the issue I noticed was unrelated to this change. This does correctly fix #21457. We'll handle the other issue in a separate PR.
